src/main/java/backend/resource/MultiModel.java

```diff
@@ -207,9 +207,12 @@ private void preprocessUpdatedIssues(List<Model> newModels) {
                     // TODO O(n^2), optimise by preprocessing into a map or sorting
                     if (!existingModel.getIssueById(i).equals(model.getIssueById(i))) {
                         assert model.getIssueById(i).isPresent();
-                        // It's no longer currently read, but it retains its updated time.
-                        // No changes to preferences.
-                        model.getIssueById(i).get().setIsCurrentlyRead(false);
+                        /** If the issue was marked read after the last update, but update came in later
+                         * due to network latency, ensure the issue retains its read state
+                        **/
+                        if (model.getIssueById(i).get().getUpdatedAt().isAfter(model.getIssueById(i).get().getMarkedReadAt().get())) {
+                            model.getIssueById(i).get().setIsCurrentlyRead(false);
+                        }
```

Comment:
Shouldn't there be some time check? The situation in concern happens because the update arrives a few seconds after marking as read. If it arrived much later, it should be recognized as a legitimate update.